client/extensions/woocommerce/app/products/product-form-details-card.js

```diff
@@ -38,7 +45,24 @@ export default class ProductFormDetailsCard extends Component {
 	// into a general purpose Higher Order Component
 	setName( e ) {
 		const { product, editProduct } = this.props;
-		editProduct( product, { name: e.target.value } );
+		const name = e.target.value;
+		editProduct( product, { name } );
+
+		if ( this.state.updateSkuOnNameChange ) {
+			const sku = trim( name ).toLowerCase().replace( /\s+/g, '-' );
+			editProduct( product, { sku } );
+		}
+	}
+
+	setSku = ( sku ) => {
+		const { product, editProduct } = this.props;
+		editProduct( product, { sku } );
```

Comment:
Yeah that's OK. WooCommerce can handle other SKUs formatted in different ways. The default ones we are generating based off name will be slug like format.